modules/generative-openai/clients/openai.go

```diff
@@ -39,7 +40,10 @@ func buildUrlFn(isLegacy bool, resourceName, deploymentID string) (string, error
 		queryParam := "api-version=2023-03-15-preview"
 		return fmt.Sprintf("%s/%s?%s", host, path, queryParam), nil
 	}
-	host := "https://api.openai.com"
+	host, ok := os.LookupEnv("OPENAI_BASE_URL")
```

Comment:
Please move looking up of `OPENAI_BASE_URL` to [module.go](https://github.com/weaviate/weaviate/blob/stable/v1.19/modules/generative-openai/module.go#L68-L71) and pass it to the `clients.New(...)` method so that you could pass it `buildUrl(...)` method